pallets/stellar-relay/src/lib.rs

```diff
@@ -0,0 +1,320 @@
+	#[pallet::event]
+	#[pallet::generate_deposit(pub (super) fn deposit_event)]
+	pub enum Event<T: Config> {}
+
+	// Errors inform users that something went wrong.
+	#[pallet::error]
+	pub enum Error<T> {
+		Base64DecodeError,
+		BoundedVecCreationFailed,
+		EnvelopeSignedByUnknownValidator,
+		InvalidExternalizedMessages,
+		InvalidEnvelopeSignature,
+		InvalidScpPledge,
+		InvalidTransactionSet,
+		InvalidTransactionXDR,
+		InvalidQuorumSetNotEnoughOrganizations,
+		InvalidQuorumSetNotEnoughValidators,
+		TransactionNotInTransactionSet,
+		TransactionSetHashMismatch,
+		TransactionSetHashCreationFailed,
+		ValidatorLimitExceeded,
+		OrganizationLimitExceeded,
+	}
+
+	#[pallet::storage]
+	#[pallet::getter(fn organizations)]
+	pub type Organizations<T: Config> =
+		StorageValue<_, BoundedVec<Organization, T::OrganizationLimit>, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn validators)]
+	pub type Validators<T: Config> =
+		StorageValue<_, BoundedVec<Validator, T::ValidatorLimit>, ValueQuery>;
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T: Config> {
+		pub validators: Vec<Validator>,
+		pub organizations: Vec<Organization>,
+		phantom: PhantomData<T>,
+	}
+
+	#[cfg(feature = "std")]
+	impl<T: Config> Default for GenesisConfig<T> {
+		fn default() -> Self {
+			Self { validators: vec![], organizations: vec![], phantom: Default::default() }
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+		fn build(&self) {
+			let validator_vec =
+				BoundedVec::<Validator, T::ValidatorLimit>::try_from(self.validators.clone());
+			assert!(validator_vec.is_ok());
+			Validators::<T>::put(validator_vec.unwrap());
+
+			let organization_vec = BoundedVec::<Organization, T::OrganizationLimit>::try_from(
+				self.organizations.clone(),
+			);
+			assert!(organization_vec.is_ok());
+			Organizations::<T>::put(organization_vec.unwrap());
+		}
+	}
+
+	// Extrinsics
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// This extrinsic is used to update/replace the current sets of validators and
+		/// organizations
+		#[pallet::weight(<T as Config>::WeightInfo::update_tier_1_validator_set())]
+		pub fn update_tier_1_validator_set(
+			origin: OriginFor<T>,
+			validators: Vec<Validator>,
+			organizations: Vec<Organization>,
+		) -> DispatchResult {
+			// Limit this call to root
+			let _ = ensure_root(origin)?;
+
+			// Ensure that the number of validators does not exceed the limit
+			ensure!(
+				validators.len() as u32 <= T::ValidatorLimit::get(),
+				Error::<T>::ValidatorLimitExceeded
+			);
+			// Ensure that the number of organizations does not exceed the limit
+			ensure!(
+				organizations.len() as u32 <= T::OrganizationLimit::get(),
+				Error::<T>::OrganizationLimitExceeded
+			);
+
+			let validator_vec = BoundedVec::<Validator, T::ValidatorLimit>::try_from(validators)
+				.map_err(|_| Error::<T>::BoundedVecCreationFailed)?;
+			Validators::<T>::put(validator_vec);
```

Comment:
I forgot what the `put` does. 

Let's say the  `ValidatorLimit` is 10,  and we already have 8 in the `Validators` storage.
Then the user wants to update 5 _new_ validators (meaning keys that do not exist in `Validators`). Does this mean the 8 existing validators are removed and _replaced_ with the 5 new ones?

Reply:
It _sets_ the value i.e. the existing values are irrelevant and replaced every time the extrinsic is used. I updated the `update_tier_1_validator_set_works()` to better illustrate this behaviour.